### PR TITLE
Hours of Operation: use current weekday, and make date configurable

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes

**Asana Ticket:** [🔺 GL-E displaying Medford/Tufts times even though service hasn't started](https://app.asana.com/0/555089885850811/1203484100934127/f)

* **First commit** basically does what the ticket asks. It checks whether the current day of the week is a weekday (value of 1 through 5) before using either it or next Monday (as it was before) as the "weekday" date for the API request.
* **Second commit** is some cleanup I found while testing the first commit. That specific test was failing for me, and rather than muck around with `Pollution.VG` internals, it seemed like a better idea to me to make that dependency redundant.
* **Third commit** Enables using `date=YYYY-MM-DD` in the URL to select the base date for requesting the Hours of Operation. It's admittedly mostly there to help me evaluate the...
* **Fourth commit** fixes the caching issue. It turns out there is a nascent caching issue! Because it was caching only by route ID, that means we wouldn't show schedule changes that happen on later dates. So that next week's visitors get the latest and greatest schedules, I added the date to the cache key.
---

#### General checks
* [x] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [x] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

### CODE REVIEW

**Comment by thecristen** [comment]
Some comparisons (old dev-blue on left, this PR on right)

### today or with `date=2022-12-09`

Can see premature Medford/Tufts schedule omitted in the updated page.

image

### with `date=2022-12-12`

Shows future schedules as expected.

image

**Comment by thecristen** [comment]
Green Line E is updated to show service to Lechmere rather than to Medford/Tufts.

image

**Review by lund5000 — APPROVED** [review]
Deploy to dev-blue failed, but it looks good from the screenshots and I love the addition of the query param for testing and debugging. ✨

File: apps/site/assets/ts/hooks/useHoursOfOperation.ts [hunk 1]
```diff
@@ -5,7 +5,11 @@ import { RapidTransitHours, TransitHours } from "../__v3api";
 const fetchData = async (
   routeIdString: string
 ): Promise<RapidTransitHours | TransitHours> =>
-  fetchJsonOrThrow(`/schedules/${routeIdString}/line/hours`);
+  fetchJsonOrThrow(
+    `/schedules/${routeIdString}/line/hours?${window.location.search.substring(
```
[the remark this answers is not in the document]
Comment (thecristen) [reply apps/site/assets/ts/hooks/useHoursOfOperation.ts:9]: [`window.location`](https://developer.mozilla.org/en-US/docs/Web/API/Location) represents the current URL, and the `search` property of that retrieves the part of the URL after and including `?` – the example [here](https://developer.mozilla.org/en-US/docs/Web/API/Location#examples) is pretty illustrative.

**Comment by thecristen** [comment]
@lund5000 thanks for looking! The branch is now up on dev-blue as well. 

re: the URL parameter: it's something Dotcom added well before my time! its support across the website varies. I think over the years later Dotcom dev teams forgot to take the date URL parameter into account when developing features. So adding `?date=2022-12-12` will work on the timetable _great_, but does squat for the line diagram. :pensive:

**Pull request merged**